mne_bids/sidecar_updates.py

```diff
@@ -136,3 +137,72 @@ def _update_sidecar(sidecar_fname, key, val):
     sidecar_json[key] = val
     with open(sidecar_fname, 'w', encoding='utf-8') as fout:
         json.dump(sidecar_json, fout)
+
+
+def update_anat_landmarks(bids_path, landmarks):
+    """Update the anatomical landmark coordinates of an MRI scan.
+
+    This will change the ``AnatomicalLandmarkCoordinates`` entry in the
+    respective JSON sidecar file, or create it if it doesn't exist.
+
+    Parameters
+    ----------
+    bids_path : mne_bids.BIDSPath
+        Path of the MR image.
+    landmarks : mne.channels.DigMontage
+        An :class:`mne.channels.DigMontage` instance with coordinates for the
+        nasion and left and right pre-auricular points in MRI voxel
+        coordinates.
+
+        .. note:: :func:`mne_bids.get_anat_landmarks` provides a convenient and
+                  reliable way to generate the landmark coordinates in the
+                  required coordinate system.
+
+    Notes
+    -----
+    .. versionadded:: 0.8
+    """
+    _validate_type(item=bids_path, types=BIDSPath, item_name='bids_path')
+    _validate_type(item=landmarks, types=DigMontage, item_name='landmarks')
+
+    positions = landmarks.get_positions()
+    coord_frame = positions['coord_frame']
+    if coord_frame != 'mri_voxel':
+        raise ValueError(
+            f'The landmarks must be specified in MRI voxel coordinates, but '
+            f'provided DigMontage is in "{coord_frame}"')
+
+    # Extract the cardinal points
+    name_to_coords_map = {
+        'LPA': positions['lpa'],
+        'NAS': positions['nasion'],
+        'RPA': positions['rpa']
+    }
+
+    # Check if coordinates for any cardinal point are missing, and convert to
+    # a list so we can easily store the data in JSON format
+    missing_points = []
+    for name, coords in name_to_coords_map.copy().items():
```

Comment:
why the copy here? I think the line below will modify still the original name_to_coords_map

Reply:
Because without the copy, inside the loop I'd modifying the object I'm iterating over, which sometimes has weird side effects. Not sure if the copy is necessary in this specific case here, though? WDYT?

Reply:
so you want name_to_coords_map to be modified outside of the function? if so I think the copy() is not necessary

Reply:
No, we're modifying `name_to_coords_map` here in this loop (in the `else` branch).
I'm not sure if this is "okay" to do while looping over `name_to_coords_map`?

Reply:
But maybe it doesn't matter as long as we're not deleting keys